utils/zerovec/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Zero-copy vector abstractions over byte arrays.
+//!
+//! This crate has two main types:
+//!
+//! - `ZeroVec<T>` for fixed-width types like `u32`
+//! - `VarZeroVec<T>` for variable-width types like `str`
+//!
+//! Both are intended as drop-in replacements for `Vec<T>` in Serde structs serialized with a
+//! format supporting a borrowed byte buffer, like Bincode.
+//!
+//! # Features
+//!
+//! - `serde`: enables Serde Serialize/Deserialize impls for ZeroVec and VarZeroVec.
+//!
+//! # Example
+//!
+//! ```
+//! # #[cfg(feature = "serde")] {
+//! use zerovec::ZeroVec;
+//!
+//! // This example requires the "serde" feature
+//! #[derive(serde::Serialize, serde::Deserialize)]
+//! pub struct DataStruct<'s> {
+//!     #[serde(borrow)]
+//!     nums: ZeroVec<'s, u32>,
+//! }
+//!
+//! let data = DataStruct {
+//!     nums: ZeroVec::from_aligned(&[211, 281, 421, 461]),
+//! };
+//! let bincode_bytes = bincode::serialize(&data)
+//!     .expect("Serialization should be successful");
+//! assert_eq!(24, bincode_bytes.len());
+//!
+//! let deserialized: DataStruct = bincode::deserialize(&bincode_bytes)
+//!     .expect("Deserialization should be successful");
+//! assert_eq!(Some(211), deserialized.nums.first());
+//! assert!(matches!(deserialized.nums, ZeroVec::Borrowed(_)));
+//! # } // feature = "serde"
+//! ```
+
+pub mod samples;
```

Comment:
`request`: can we only have it in debug mode or behind a flag?

I wish Rust had a better solution because I know it's not exactly trivial to have `samples` be accessible by tests and benchmarks and examples all at once, but having `samples` be in the public API just because we need resources to test things on seems like giving up on clean public API design.
I'd love to stop that trend before it takes over.

Reply:
I want to share these resource between test and bench.

Bench is built without debug mode and debug assertions, so `#[cfg(debug_assertions)]` doesn't work here.

I can add a new feature like `"test"` or something, but I don't want to do that unless we decide to adopt that policy across ICU4X.

Follow-up: #657